### ISSUE TITLE
Ability to set Headers for function.call

### ISSUE THREAD

**Issue by NickFranceschina** [n6g7/redux-saga-firebase#95]
implementing the "authorized http function endpoint" example and realizing I can't set headers when calling `rsf.functions.call(...)` 

https://github.com/firebase/functions-samples/tree/master/authorized-https-endpoint

I can do a normal `fetch` for now

**Comment by n6g7** [comment]
Hi @NickFranceschina,

That's good feedback, thanks.

I think this can easily be addressed by duplicating the [`fetch()` api](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) and adding an optional `init` parameter to the [`functions.call`](https://github.com/n6g7/redux-saga-firebase/blob/master/src/functions.js#L14) method. We'd pass this parameter down as is to `fetch` and we could write a custom function call like so:

```js
yield call(rsf.functions.call, 'myFunction', {
  param1: 1,
  param2: 2
}, {
  method: 'POST',
  headers: {
    'Authorization': 'Bearer <token>'
  }
})
```

What do you think?

**Comment by NickFranceschina** [comment]
sounds perfect!  I would have offered a PR but I'm so swamped right now!!

**Comment by n6g7** [comment]
No problem, I'll do that this weekend if we don't get a PR before then. :)

**Comment by n6g7** [comment]
@NickFranceschina I released [v0.10.0](https://github.com/n6g7/redux-saga-firebase/releases/tag/v0.10.0) with this new argument, let me know if it solves your problem. :)

**Comment by NickFranceschina** [comment]
very cool thank you!
